cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/testdata/gcloud-setup.md

```diff
@@ -0,0 +1,151 @@
+date: Fri, 26 Aug 2022 02:18:15 GMT
+content-length: 0
+content-type: text/html; charset=UTF-8
+```
+
+2. check [catalog API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-repositories) is working 
+  * with service account access token
+```
+$ curl -iL https://us-west1-docker.pkg.dev/v2/_catalog -H "Authorization: Bearer $GCP_TOKEN"
+  HTTP/2 200
+  content-type: application/json; charset=utf-8
+  docker-distribution-api-version: registry/2.0
+  date: Wed, 24 Aug 2022 16:10:18 GMT
+  content-length: 75
+
+  {"repositories":["vmware-kubeapps-ci/stefanprodan-podinfo-clone/podinfo"]}
+```
+  * with JSON key file
+```
+$ curl -iL https://us-west1-docker.pkg.dev/v2/_catalog -H "$(oauth2l header --scope=cloud-platform)"
+HTTP/2 200 
+content-type: application/json; charset=utf-8
+docker-distribution-api-version: registry/2.0
+date: Fri, 26 Aug 2022 02:22:52 GMT
+content-length: 75
+
+{"repositories":["vmware-kubeapps-ci/stefanprodan-podinfo-clone/podinfo"]}
+```
+
+3. check flux is able to reconcile a `HelmRepository` and a `HelmRelease`:
+  * with service account access tokens (short-lived)
+```
+$ kubectl create secret docker-registry gcp-repo-auth-9 \
+  --docker-server=us-west1-docker.pkg.dev \
+  --docker-username=oauth2accesstoken \
+  --docker-password="$(gcloud auth application-default print-access-token)"
+
+$ flux create source helm podinfo-9 \
+       --url=oci://us-west1-docker.pkg.dev/vmware-kubeapps-ci/stefanprodan-podinfo-clone \
+       --namespace=default \
+       --secret-ref=gcp-repo-auth-9
+
+$ flux create hr podinfo-9 \
+      --source=HelmRepository/podinfo-9.default \
+      --chart=podinfo
+✚ generating HelmRelease
+► applying HelmRelease
+✔ HelmRelease created
+◎ waiting for HelmRelease reconciliation
+✔ HelmRelease podinfo-9 is ready
+✔ applied revision 6.1.8
+```
+  * with JSON key file (long-lived)
+```
+kubectl create secret docker-registry gcp-repo-auth \
+  --docker-server=us-west1-docker.pkg.dev \
+  --docker-username=_json_key \
+  --docker-password="$(cat $GOOGLE_APPLICATION_CREDENTIALS)"
+
+$ flux create source helm podinfo-7 \
+       --url=oci://us-west1-docker.pkg.dev/vmware-kubeapps-ci/stefanprodan-podinfo-clone \
+       --namespace=default \
+       --secret-ref=gcp-repo-auth
+✚ generating HelmRepository source
+► applying HelmRepository source
+✔ source created
+◎ waiting for HelmRepository source reconciliation
+✔ HelmRepository source reconciliation completed
+
+$ flux create hr podinfo-7 \
+>      --source=HelmRepository/podinfo-7.default \
+>      --chart=podinfo
+✚ generating HelmRelease
+► applying HelmRelease
+✔ HelmRelease updated
+◎ waiting for HelmRelease reconciliation
+✔ HelmRelease podinfo-7 is ready
+✔ applied revision 6.1.8
+```
+
+## Notes
+1. After a cold start (several hours of not running the tests), running the test always first at first
+```
+$ go test -v -timeout 9999s -run TestKindClusterAvailablePackageEndpointsForOCI 
```

Comment:
Thanks for documenting it.
Do you think we could enable this tests if we add to CI the right auth env variables (tokens, etc.)? For forked repositories tests wouldn't apply.
For example, I'm wondering about concurrent runs.

Reply:
yes, I do, but it would take a bit of work. It's not just limited environment variables. There are other issues too work out. For example, with GitHub one particular issue is that [there is no automated way](https://github.com/cli/cli/discussions/6003) to set the package visibility to public with gh CLI, so my current script needs web browser interaction to do it at the time test bed is created.  There are a bunch of CLIs to install on CI server too. There are other pesky issues too, but it all can be worked out given a bit of time
BTW, Antonio already asked me this a few weeks ago: https://github.com/vmware-tanzu/kubeapps/pull/5178#pullrequestreview-1066619644. My answer is the same